src/databricks/labs/ucx/installer/logs.py

```diff
@@ -0,0 +1,149 @@
+import datetime
+import logging
+import re
+from collections.abc import Iterator
+from dataclasses import dataclass
+from pathlib import Path
+from typing import TextIO
+
+from databricks.labs.lsql.backends import SqlBackend
+
+from databricks.labs.ucx.framework.crawlers import CrawlerBase
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class LogRecord:
+    ts: int  # fully specified timestamp in UTC (offset from a workspace fs file creation timestamp)
+    job_id: int  # determined from workspace file path, resolved from the install state
+    workflow_name: str
+    task_name: str
+    job_run_id: int
+    level: str
+    component: str
+    message: str
+
+
+@dataclass
+class _PartialLogRecord:
+    """The information found within a log file record."""
+    hour: str
+    minute: str
+    second: str
+    level: str
+    component: str
+    message: str
+
+
+def _get_task_names_at_runtime(log_path: Path) -> list[str]:
+    log_files = log_path.glob("*.log")
```

Comment:
we're using `TimedRotatingFileHandler` (see https://docs.python.org/3/library/logging.handlers.html#logging.handlers.TimedRotatingFileHandler), so this glob is incomplete